wdl-engine/src/lib.rs

```diff
@@ -10,10 +10,267 @@ mod stdlib;
+        .context("failed to find document in analysis results")?;
+    let document = result.document();
+
+    let mut engine = Engine::new(LocalTaskExecutionBackend::new());
+    let (path, name, inputs) = if let Some(path) = inputs {
+        let abs_path = absolute(&path).with_context(|| {
+            format!(
+                "failed to determine the absolute path of `{path}`",
+                path = path.display()
+            )
+        })?;
+        match Inputs::parse(document, &abs_path)? {
+            Some((name, inputs)) => (Some(path), name, inputs),
+            None => bail!(
+                "inputs file `{path}` is empty; use the `--name` option to specify the name of \
+                 the task or workflow to run",
+                path = path.display()
+            ),
+        }
+    } else if let Some(name) = name {
+        if document.task_by_name(&name).is_some() {
+            (None, name, Inputs::Task(Default::default()))
+        } else if document.workflow().is_some() {
+            (None, name, Inputs::Workflow(Default::default()))
+        } else {
+            bail!("document does not contain a task or workflow named `{name}`");
+        }
+    } else {
+        let mut iter = document.tasks();
+        let (name, inputs) = iter
+            .next()
+            .map(|t| (t.name().to_string(), Inputs::Task(Default::default())))
+            .or_else(|| {
+                document
+                    .workflow()
+                    .map(|w| (w.name().to_string(), Inputs::Workflow(Default::default())))
+            })
+            .context("inputs file is empty and the WDL document contains no tasks or workflow")?;
+
+        if iter.next().is_some() {
+            bail!("inputs file is empty and the WDL document contains more than one task");
+        }
+
+        (None, name, inputs)
+    };
+
+    match inputs {
+        Inputs::Task(mut inputs) => {
+            let task = document
+                .task_by_name(&name)
+                .ok_or_else(|| anyhow!("document does not contain a task named `{name}`"))?;
+
+            // Ensure all the paths specified in the inputs file are relative to the file's
+            // directory
+            if let Some(path) = path.as_ref().and_then(|p| p.parent()) {
+                inputs.join_paths(task, path);
+            }
+
+            let mut evaluator = TaskEvaluator::new(&mut engine);
+            match evaluator
+                .evaluate(document, task, &inputs, &output, &name)
+                .await
+            {
+                Ok(evaluated) => {
+                    match evaluated.into_result() {
+                        Ok(outputs) => {
+                            // Buffer the entire output before writing it out in case there are
+                            // errors during serialization.
+                            let mut buffer = Vec::new();
+                            let mut serializer = serde_json::Serializer::pretty(&mut buffer);
```

Comment:
Can you just use [`to_string_pretty()`](https://docs.rs/serde_json/latest/serde_json/fn.to_string_pretty.html) here?

Reply:
Should be able to now that we no longer have to do custom serialization like we did before the last refactoring.